packages/@lwc/features/src/__tests__/flags.spec.ts

```diff
@@ -0,0 +1,307 @@
+            if (ENABLE_FEATURE_NULL === null) {
+                console.log('ENABLE_FEATURE_NULL === null');
+            }
+            if (isTrue(ENABLE_FEATURE_TRUE)) {
+                console.log('isTrue(ENABLE_FEATURE_TRUE)');
+            }
+            if (!ENABLE_FEATURE_FALSE) {
+                console.log('!ENABLE_FEATURE_FALSE');
+            }
+        `,
+        output: `
+            import { ENABLE_FEATURE_TRUE, ENABLE_FEATURE_FALSE, ENABLE_FEATURE_NULL, runtimeFlags } from '@lwc/features';
+
+            if (ENABLE_FEATURE_NULL === null) {
+              console.log('ENABLE_FEATURE_NULL === null');
+            }
+
+            if (isTrue(ENABLE_FEATURE_TRUE)) {
+              console.log('isTrue(ENABLE_FEATURE_TRUE)');
+            }
+
+            if (!ENABLE_FEATURE_FALSE) {
+              console.log('!ENABLE_FEATURE_FALSE');
+            }
+        `,
+    },
+    'should not transform identifiers that look like feature flags but are not imported': {
+        code: `
+            import { ENABLE_FEATURE_TRUE } from '@lwc/features';
+            const ENABLE_FEATURE_FALSE = true;
+            if (ENABLE_FEATURE_FALSE) {
+                console.log('ENABLE_FEATURE_FALSE');
+            }
+            function awesome() {
+                const ENABLE_FEATURE_TRUE = null;
+                if (ENABLE_FEATURE_TRUE) {
+                    console.log('ENABLE_FEATURE_TRUE');
+                }
+            }
+        `,
+        output: `
+            import { ENABLE_FEATURE_TRUE, runtimeFlags } from '@lwc/features';
+            const ENABLE_FEATURE_FALSE = true;
+
+            if (ENABLE_FEATURE_FALSE) {
+              console.log('ENABLE_FEATURE_FALSE');
+            }
+
+            function awesome() {
+              const ENABLE_FEATURE_TRUE = null;
+
+              if (ENABLE_FEATURE_TRUE) {
+                console.log('ENABLE_FEATURE_TRUE');
+              }
+            }
+        `,
+    },
+    'should not transform feature flags when used with a ternary operator': {
+        code: `
+            import { ENABLE_FEATURE_NULL } from '@lwc/features';
+            console.log(ENABLE_FEATURE_NULL ? 'foo' : 'bar');
+        `,
+        output: `
+            import { ENABLE_FEATURE_NULL, runtimeFlags } from '@lwc/features';
+            console.log(ENABLE_FEATURE_NULL ? 'foo' : 'bar');
+        `,
+    },
+    'should not transform nested feature flags': {
+        code: `
+            import { ENABLE_FEATURE_TRUE, ENABLE_FEATURE_NULL } from '@lwc/features';
+            if (ENABLE_FEATURE_NULL) {
+                if (ENABLE_FEATURE_TRUE) {
+                    console.log('this looks like a bad idea');
+                }
+            }
+        `,
+        output: `
+            import { ENABLE_FEATURE_TRUE, ENABLE_FEATURE_NULL, runtimeFlags } from '@lwc/features';
+
+            if (runtimeFlags.ENABLE_FEATURE_NULL) {
+              if (ENABLE_FEATURE_TRUE) {
```

Comment:
why not?

Reply:
Nested feature flags sounds very complicated so I'm not sure we want to support that. We only need to remove a single line of code if we change our minds.

Reply:
To expand a little more, the number of possible outcomes is exponential when it comes to nesting. We should avoid that unless it's really needed.